<?xml version="1.0" encoding="utf-8"?>
<!-- This URDF was automatically created by SolidWorks to URDF Exporter! Originally created by Stephen Brawner (brawner@gmail.com) 
     Commit Version: 1.6.0-4-g7f85cfe  Build Version: 1.6.7995.38578
     For more information, please see http://wiki.ros.org/sw_urdf_exporter -->

<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="rob_description">


  <xacro:include filename="$(find rob_description)/urdf/rob_plugins.gazebo"/>

  <link name="world">
  </link>

  <joint name="base_joint" type="fixed">
    <parent link="world" />
    <child link="base_link" />
    <origin xyz="0 0 0" rpy="0 0 0.17" />
  </joint>

  <link name="base_link">
    <inertial>
      <origin xyz="0.0429301108999509 -0.0510754466449302 0.0293852995706934" rpy="0 0 0" />
      <mass value="0.45432485870868" />
      <inertia ixx="0.0029066579318034" ixy="1.52586102405724E-06" ixz="1.62445178667574E-05" iyy="0.00127300003261156" iyz="2.55065572355847E-08" izz="0.00375436802461814" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/base_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0.752941176470588 0.752941176470588 0.752941176470588 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/base_link.STL" />
      </geometry>
    </collision>
  </link>
  <link name="wheel1_link">
    <inertial>
      <origin xyz="0.000138449426315122 -0.00699863165738515 0.0181270123819037" rpy="0 0 0" />
      <mass value="0.166945648008905" />
      <inertia ixx="6.72964134165798E-05" ixy="5.95216536750434E-12" ixz="3.19966084058516E-12" iyy="6.72926792752873E-05" iyz="2.2405037899089E-13" izz="0.000113206411587724" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wheel1_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wheel1_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="wheel1_joint" type="continuous">
    <origin xyz="0.011232792252951 0.0550133054637493 0.01926270910094" rpy="-1.5707963267949 0.177359227009232 0" />
    <parent link="base_link" />
    <child link="wheel1_link" />
    <axis xyz="0 0 1" />
  </joint>
  <link name="wheel2_link">
    <inertial>
      <origin xyz="0.000138449427382649 -0.0069986316601361 0.018127012381964" rpy="0 0 0" />
      <mass value="0.166945648038239" />
      <inertia ixx="6.72964134176824E-05" ixy="5.94446824189331E-12" ixz="3.16585447322871E-12" iyy="6.72926792843813E-05" iyz="2.3733916667747E-13" izz="0.000113206411597874" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wheel2_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0.792156862745098 0.819607843137255 0.933333333333333 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wheel2_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="wheel2_joint" type="continuous">
    <origin xyz="0.0112327921426415 -0.154845357288986 0.02076270910094" rpy="1.5707963267949 0.691543612587939 0" />
    <parent link="base_link" />
    <child link="wheel2_link" />
    <axis xyz="0 0 -1" />
  </joint>
  <link name="robot_arm_base_link">
    <inertial>
      <origin xyz="0.000890229377213621 0.0223982076492734 -1.51953095939295E-10" rpy="0 0 0" />
      <mass value="0.0938941939628843" />
      <inertia ixx="0.00011393791922916" ixy="-1.35515920698357E-06" ixz="-1.97717583289253E-13" iyy="0.000197548704096291" iyz="-1.90758289806404E-13" izz="0.000124247629168402" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/robot_arm_base_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0.501960784313725 0.501960784313725 0.501960784313725 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/robot_arm_base_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="robot_arm_base_joint" type="fixed">
    <origin xyz="0.0542327893849049 -0.0507628932149278 0.0582627122631441" rpy="1.5707963267949 0 -3.14159265358979" />
    <parent link="base_link" />
    <child link="robot_arm_base_link" />
    <axis xyz="0 0 0" />
  </joint>
  <link name="waist_link">
    <inertial>
      <origin xyz="-0.00705756789604821 0.0194229347811724 1.038142207338E-07" rpy="0 0 0" />
      <mass value="0.0897213187915641" />
      <inertia ixx="7.72728720700333E-05" ixy="7.96790903927782E-06" ixz="2.77140490095276E-10" iyy="7.06954475494065E-05" iyz="-6.66096656459277E-11" izz="6.69370927399034E-05" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/waist_link.STL" />
      </geometry>
      <material name="">
        <color rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/waist_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="waist_joint" type="revolute">
    <origin xyz="0 0.05 0" rpy="0 1.08100913503451 0" />
    <parent link="robot_arm_base_link" />
    <child link="waist_link" />
    <axis xyz="0 0 1" />
    <limit lower="-1.5707963267949" upper="1.5707963267949" effort="50" velocity="0.5" />
  </joint>
  <link name="shoulder_link">
    <inertial>
      <origin xyz="2.20043870011177E-06 0.0103077592658962 -1.09059368272324E-06" rpy="0 0 0" />
      <mass value="0.137970572349197" />
      <inertia ixx="3.08345354645482E-05" ixy="-2.13068535969506E-09" ixz="-6.47325294636156E-10" iyy="0.000431654970376359" iyz="1.06137375873849E-09" izz="0.000412898353495999" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/shoulder_link.STL" />
      </geometry>
      <material name="">
        <color rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/shoulder_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="shoulder_joint" type="revolute">
    <origin xyz="0 0.111215239821368 0.017928059569213" rpy="-3.14159265358955 -0.259001141232201 1.57079632679483" />
    <parent link="waist_link" />
    <child link="shoulder_link" />
    <axis xyz="0 1 0" />
    <limit lower="-2.0943951023932" upper="2.0943951023932" effort="75" velocity="0.8" />
  </joint>
  <link name="elbow_link">
    <inertial>
      <origin xyz="-0.0316987440184093 0.0105785583288961 -0.000100527970298914" rpy="0 0 0" />
      <mass value="0.0803343703233642" />
      <inertia ixx="2.12570030269286E-05" ixy="8.16587499796508E-07" ixz="-2.54261433454123E-07" iyy="0.000117225501185926" iyz="1.05325896499179E-08" izz="0.000104519029575083" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/elbow_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0 0 0.501960784313725 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/elbow_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="elbow_joint" type="revolute">
    <origin xyz="0.0700000000000071 -0.00250000000002339 0" rpy="0 0.666778469149742 -3.14159265358968" />
    <parent link="shoulder_link" />
    <child link="elbow_link" />
    <axis xyz="0 1 0" />
    <limit lower="-2.6179938779915" upper="2.6179938779915" effort="80" velocity="1.0" />
  </joint>
  <link name="wrist_link">
    <inertial>
      <origin xyz="3.46276801557743E-05 -0.00411008004478342 0.0184466953896076" rpy="0 0 0" />
      <mass value="0.0204564905574409" />
      <inertia ixx="5.09450576471333E-06" ixy="-3.90195217616513E-09" ixz="1.06590555803014E-08" iyy="6.0754095092678E-06" iyz="3.8979988594088E-07" izz="2.96028488948377E-06" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wrist_link.STL" />
      </geometry>
      <material name="">
        <color rgba="0.556862745098039 0.556862745098039 0.556862745098039 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wrist_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="wrist_joint" type="revolute">
    <origin xyz="-0.105699999999926 0.0109254046180036 8.45116508766813E-05" rpy="-1.57079632679684 -1.38849503153902 1.57079632679672" />
    <parent link="elbow_link" />
    <child link="wrist_link" />
    <axis xyz="0 0 1" />
    <limit lower="-1.5707963267949" upper="1.5707963267949" effort="60" velocity="1.2" />
  </joint>
  <link name="wrist_pitch_link">
    <inertial>
      <origin xyz="0.024485098459644 0.0047925620128958 -0.0016042523851675" rpy="0 0 0" />
      <mass value="0.0155758863589592" />
      <inertia ixx="2.584126351928E-06" ixy="-8.30847952796364E-07" ixz="5.46132239991505E-07" iyy="6.24782073073235E-06" iyz="1.00345546004568E-08" izz="4.87873061642217E-06" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wrist_pitch_link.STL" />
      </geometry>
      <material name="">
        <color rgba="1 1 1 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <mesh filename="package://rob_description/meshes/wrist_pitch_link.STL" />
      </geometry>
    </collision>
  </link>
  <joint name="wrist_pitch_joint" type="revolute">
    <origin xyz="0 -0.0168500000000001 0.0284" rpy="1.5707963267949 -0.262431797603188 0" />
    <parent link="wrist_link" />
    <child link="wrist_pitch_link" />
    <axis xyz="0 0 1" />
    <limit lower="-1.5707963267949" upper="1.5707963267949" effort="50" velocity="1.0" />
  </joint>
  <link name="camera_link">
    <inertial>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <mass value="0.1" />
      <inertia ixx="0.0001" ixy="0" ixz="0" iyy="0.0001" iyz="0" izz="0.0001" />
    </inertial>
    <visual>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.03 0.02 0.02" />
      </geometry>
      <material name="blue">
        <color rgba="0 0 1 1" />
      </material>
    </visual>
    <collision>
      <origin xyz="0 0 0" rpy="0 0 0" />
      <geometry>
        <box size="0.03 0.02 0.02" />
      </geometry>
    </collision>
  </link>
  <joint name="camera_joint" type="fixed">
    <origin xyz="0.1 0 0.05" rpy="0 0 0" />
    <parent link="base_link" />
    <child link="camera_link" />
  </joint>

  <!-- <transmission name="robot_arm_base_link">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="robot_arm_base_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="robot_arm_base_link">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="waist_link">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="waist_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="waist_link">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="shoulder_link">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="shoulder_link">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="elbow_link">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="elbow_link">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="wrist_link">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_link">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

  <transmission name="wrist_pitch_link">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_pitch_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="wrist_pitch_link">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>

 -->


</robot>